src/util/simplify_expr.cpp

```diff
@@ -153,6 +158,148 @@ simplify_exprt::simplify_popcount(const popcount_exprt &expr)
   return unchanged(expr);
 }
 
+/// Simplify String.endsWith function when argument are constant
+/// \param expr [in,out]: the expression to simplify
+/// \param ns: namespace
+/// \return: true iff expr was not modified
+static simplify_exprt::resultt<> simplify_string_endswith_func(
+  const function_application_exprt &expr,
+  const namespacet &ns)
+{
+  const function_application_exprt &function_app =
+    to_function_application_expr(expr);
+
+  const refined_string_exprt &s1 =
+    to_string_expr(function_app.arguments().at(0));
+
+  const auto s1_data_opt = try_get_string_data_array(s1, ns);
+
+  if(!s1_data_opt)
+  {
+    return simplify_exprt::unchanged(expr);
+  }
+
+  const array_exprt &s1_data = s1_data_opt->get();
+
+  const refined_string_exprt &s2 =
+    to_string_expr(function_app.arguments().at(1));
+
+  const auto s2_data_opt = try_get_string_data_array(s2, ns);
+
+  if(!s2_data_opt)
+  {
+    return simplify_exprt::unchanged(expr);
+  }
+
+  const array_exprt &s2_data = s2_data_opt->get();
+
+  const bool res = s2_data.operands().size() <= s1_data.operands().size() &&
+                   std::equal(
+                     s2_data.operands().rbegin(),
+                     s2_data.operands().rend(),
+                     s1_data.operands().rbegin());
+
+  return from_integer(res ? 1 : 0, expr.type());
+}
+
+/// Simplify String.isEmpty function when arguments is constant
+/// \param expr [in,out]: the expression to simplify
+/// \param ns: namespace
+/// \return: true iff expr was not modified
+static simplify_exprt::resultt<> simplify_string_is_empty(
+  const function_application_exprt &expr,
+  const namespacet &ns)
+{
+  const function_application_exprt &function_app =
+    to_function_application_expr(expr);
+
+  const refined_string_exprt &s =
+    to_string_expr(function_app.arguments().at(0));
+
+  if(s.length().id() != ID_constant)
+  {
+    return simplify_exprt::unchanged(expr);
+  }
+
+  const auto numeric_length =
+    numeric_cast<mp_integer>(to_constant_expr(s.length()));
+
+  if(!numeric_length.has_value())
+  {
+    return simplify_exprt::unchanged(expr);
+  }
```

Comment:
Could this become an invariant `INVARIANT(numeric_length.has_value(), ...)` instead?

Reply:
followed @romainbrenguier's suggestion